bridge/engine/EngineMixer.cpp

```diff
@@ -1839,14 +1886,40 @@ void EngineMixer::onRtcpPacketDecoded(transport::RtcTransport* sender,
 SsrcOutboundContext* EngineMixer::obtainOutboundSsrcContext(size_t endpointIdHash,
     concurrency::MpmcHashmap32<uint32_t, SsrcOutboundContext>& ssrcOutboundContexts,
     const uint32_t ssrc,
-    const RtpMap& rtpMap)
+    const RtpMap& rtpMap,
+    const bool ssrcRewrite)
 {
     auto ssrcOutboundContextItr = ssrcOutboundContexts.find(ssrc);
     if (ssrcOutboundContextItr != ssrcOutboundContexts.cend())
     {
+        if (ssrcOutboundContextItr->second.markedForDeletion)
+        {
+            // The outboundContext is marked for deletion after owner inbound stream
+            // as been removed (only for non ssrcRewrite). This means the job to
+            // remove OutboundSsrc has already been enqueued and it will be deleted
+            // soon. Then we MUST NOT enqueue any more jobs with a reference to this
+            // SsrcOutboundContext
+            assert(!ssrcRewrite);
+            return nullptr;
+        }
+
         return &ssrcOutboundContextItr->second;
     }
 
+    if (!ssrcRewrite)
+    {
+        // When is not ssrc rewrite, we have to ensure that ssrcInboundContext is not been
+        // decommissioned already. If it was decommissioned and the ssrcOutboundContext
+        // is not in the ssrcOutboundContexts, most likely the reason is because we are
+        // processing an old packet in the queue and the ssrcOutboundContext has been deleted already.
+        // In such case we we will not recreate the outboundContext because it would be
+        // deleted only in the end of the conference (keep with it all associated resources, like video caches)
+        if (!_ssrcInboundContexts.contains(ssrc))
+        {
+            return nullptr;
```

Comment:
I think this never happens, but f the SsrcInboundContext does not exist (means it was decommissed) is because we are processing a packet for a stream that was removed recently so we will not create an outboundContext because most likely it was deleted recently also. Moreover, if inbound stream was removed and we create an OutboundContext again, it would be removed only when the oubound stream was removed.

This should never hit here when SsrcInboundContext id decommissed by timeout. Otherwise, it would mean that packet is more than 5 minutes old